flixel/animation/FlxAnimationController.hx

```diff
@@ -777,6 +777,55 @@ class FlxAnimationController implements IFlxDestroyable
 		play(AnimName);
 		return AnimName;
 	}
+		
+	public function animationList():Array<FlxAnimation>
+	{
+		var arrayNames:Array<FlxAnimation> = [];
+		
+		for (i in _animations)
+		{
+			arrayNames.push(i);
+		}
+
+		return arrayNames;
+	}
+
+	public function nameList():Array<String>
+	{
+		var anims = animationList();
+		var names:Array<String> = [];
+		for (i in 0...anims.length)
+		{
+			names.push(anims[i].name);
+		}
+
+		return names;
+	}
+
+	public function exists(name:String):Bool
+	{
+		return nameList().contains(name);
+	}
+	
+	public function renameAnimation(name:String, value:String)
+	{
+		var prevAnims = animationList();
+
+		_animations = new Map<String, FlxAnimation>();
+
+		for (i in 0...prevAnims.length)
+		{
+			if (prevAnims[i].name == name)
+			{
+				prevAnims[i].name = value;
+				_animations.set(value, prevAnims[i]);
+			}
+			else
+			{
+				_animations.set(prevAnims[i].name, prevAnims[i]);
+			}
+		}
```

Comment:
1. we don't need to create a new animation map just to rename 1 entry, maps have a remove function. we also don't need to loop through animations, we can access by name via the map
2. we should throw a warning when they try to rename a non-existent animation.
3. We should return a bool indicating whether it was successful.

putting it all together, something like:
```hx
var anim = _animations[oldName];
if (anim == null)
{
	FlxG.log.warn('No animation called "$newName"');
	return false;
}
_animations[newName] = anim;
anim.name = newName;
_animations.remove(oldName);
return true;
```

Reply:
I changed the code a bit but I think it gives the same thing and it gives some nice good-looking code imo.
Resolved